src/taco-utils/npmHelper.ts

```diff
@@ -0,0 +1,50 @@
+/**
+﻿ *******************************************************
+﻿ *                                                     *
+﻿ *   Copyright (C) Microsoft. All rights reserved.     *
+﻿ *                                                     *
+﻿ *******************************************************
+﻿ */
+
+/// <reference path="../typings/node.d.ts" />
+/// <reference path="../typings/npm.d.ts" />
+/// <reference path="../typings/q.d.ts" />
+
+"use strict";
+
+import npm = require("npm");
+import path = require("path");
+import installLogLevel = require("./installLogLevel");
+import Q = require("q");
+
+import InstallLogLevel = installLogLevel.InstallLogLevel;
+
+module TacoUtility {
+	export class NpmHelper {
+		private static runNpmCommand(npmCommand: string, path: string, args: string[], optionalArgs: any, logLevel?: InstallLogLevel): Q.Promise<any> {
```

Comment:
Since this method is private and you always call it with logLevel (possibly undefined, but provided) but not always with optionalArgs, might be worth reordering the arguments.